drivers/sensor/wsen_pdus/wsen_pdus.c

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (c) 2023 Würth Elektronik eiSos GmbH & Co. KG
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT we_wsen_pdus
+
+#include <stdlib.h>
+
+#include <zephyr/sys/__assert.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/logging/log.h>
+
+#include "wsen_pdus.h"
+
+LOG_MODULE_REGISTER(WSEN_PDUS, CONFIG_SENSOR_LOG_LEVEL);
+
+static int pdus_sample_fetch(const struct device *dev, enum sensor_channel chan)
+{
+	const struct pdus_config *const config = dev->config;
+	struct pdus_data *data = dev->data;
+	float pressure;
+	float temperature;
+
+	__ASSERT_NO_MSG(chan == SENSOR_CHAN_ALL);
+
+	if (PDUS_getPressureAndTemperature_float(&data->sensor_interface,
+						 config->sensor_type, &pressure,
+						 &temperature) != WE_SUCCESS) {
+		LOG_ERR("Failed to fetch data sample");
+		return -EIO;
+	}
+
+	data->pressure_k_pa = pressure;
+	data->temperature_deg_c = temperature;
+
+	return 0;
+}
+
+static int pdus_channel_get(const struct device *dev, enum sensor_channel chan,
+			    struct sensor_value *value)
+{
+	struct pdus_data *data = dev->data;
+
+	if (chan == SENSOR_CHAN_PRESS) {
+		value->val1 = (int32_t)data->pressure_k_pa;
+		value->val2 = (((int32_t)(data->pressure_k_pa * 1000)) % 1000) * (1000000 / 1000);
```

Comment:
Why not: 
```value->val2 = (((int32_t)(data->pressure_k_pa * 1000)) % 1000) * 1000;```

Am I missing something?

Reply:
hello @avisconti ,
thank you for the review. The code is written in a way that the data conversion is conform to the other WSEN drivers and easy to unterstand. Nevertheless, you are right, that this piece of code can be simplified. I will do this and push again this afternoon.
Please do not hesitate to remind me of more improvements. Thank you!
BR Matthias